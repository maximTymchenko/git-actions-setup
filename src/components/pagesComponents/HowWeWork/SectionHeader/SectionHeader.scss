@import '../../../../styles/vars';

.section-header__wrapper {
    @media screen and (max-width: $mobile) {
        max-width: 360px;
        margin-bottom: 30px;
    }
}
