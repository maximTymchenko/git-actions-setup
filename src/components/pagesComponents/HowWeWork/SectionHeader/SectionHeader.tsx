import React, { FC } from 'react';
import TextDescription from '../../../common/TextDescription';
import MainTitle from '../../../common/MainTitle';
import SectionTitle from '../../../common/SectionTitle';
import './SectionHeader.scss';

interface ISectionHeader {
    subtitleTextF: string;
    subtitleTextB: string;
    subtitleTextS: string;
    secondaryTitleText: string;
    titleText: string;
    large: string;
    description: string;
    descriptionMain: string;
    descriptionPrimary: string;
    descriptionSecondary: string;
    typeValue: string;
    subtype: string;
    size: string;
}
const SectionHeader: FC = (props: ISectionHeader) => {
    return (
        <div data-testid="section-header" className="section-header__wrapper">
            <MainTitle
                subtypeB="bold"
                subtype="small"
                type={props.typeValue}
                subtitleF={props.subtitleTextF}
                subtitleS={props.subtitleTextS}
                subtitleB={props.subtitleTextB}
                title={props.titleText}
            />
            <SectionTitle type="primary" className="title" title={props.secondaryTitleText} />
        </div>
    );
};

export default SectionHeader;
