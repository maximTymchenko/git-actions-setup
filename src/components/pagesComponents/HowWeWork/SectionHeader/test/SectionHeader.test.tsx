import React from 'react'
import SectionHeader from '../SectionHeader'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders elem correctly', () => {
  const { getByTestId } = render(
    <SectionHeader
      descriptionMain="test"
      descriptionPrimary="test"
      descriptionSecondary="test"
      secondaryTitleText="test"
    ></SectionHeader>
  )
  expect(getByTestId('section-header')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer
    .create(
      <SectionHeader
        descriptionMain="test"
        descriptionPrimary="test"
        descriptionSecondary="test"
        secondaryTitleText="test"
      ></SectionHeader>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
