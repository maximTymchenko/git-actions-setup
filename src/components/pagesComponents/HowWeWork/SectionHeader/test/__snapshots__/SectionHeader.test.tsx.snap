// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="section-header__wrapper"
  data-testid="section-header"
>
  <div>
    <div
      className="title__wrapper"
    >
      <h2
        className="title title--small"
        data-testid="section-title"
      />
      <h2
        className="title title--bold"
        data-testid="section-title"
      />
      <h2
        className="title title--small"
        data-testid="section-title"
      />
    </div>
    <h1
      className="title title--undefined"
      data-testid="main-title"
    />
  </div>
  <h2
    className="title title--primary"
    data-testid="section-title"
  >
    test
  </h2>
</div>
`;
